system/mp/mpcomm.cpp

```diff
@@ -826,11 +826,22 @@ protected: friend class CMPPacketReader;
 #ifdef _TRACE
                                 EXCLOG(e, "MP: Failed to connect");
 #endif
-                                e->Release();
                                 if ((retrycount--==0)||(tm.timeout==MP_ASYNC_SEND))
                                 {   // don't bother retrying on async send
+                                    e->Release();
```

Comment:
minor: I'd have been tempted to add a Owned<IException> exception = e; at the head of the catch block, to avoid the need for individual e->Release(0's lines in the right places here and below.